### DISCUSSION TITLE
Change how automatic tags are determined

### DISCUSSION THREAD

**Discussion by OSAwebsite** [code4recovery/12-step-meeting-list#1206] [Ideas]
Hi team,
When I upload a brand new meeting or a re-activated one, I use the import sheet rather than manual entry.
Not too long ago I had to refresh my entire list of meetings.
I have discovered that some incorrectly got the tag 'temporarily closed'.
When inspecting the listing, I saw that the tag 'Can I currently attend this meeting in person?' was set to 'no' and this was unexpected.
When I checked my database, I found what caused the error. I had the checkbox 'Location temporarily closed' ticked. This was an error on my part.

However...

This checkbox is nowhere to be found in the backend of a meeting listing. So it is easy to miss an error like that on the website listing. When I do a complete refresh, I do scroll through the list of all meetings but this time I missed about 6 meetings that got the red tag 'temporarily closed'.

The only way to correct this was to manually reset the 'Can I currently attend this meeting in person?' from 'No' to 'yes'.

My request is this:
is it possible to add the 'Location temporarily closed' box to the backend so I will at least be alerted quicker to a mistake?

As a general comment I think that there is some implicit coding going on in the background that make it too easy to miss a mistake.

Another example is the link to a Zoom meeting. It depends on whether or not there is an URL entered for the online meeting.
Would it not be much easier to have 'yes/no' toggles that can be included in the import sheet?

Yes, the meeting can be attended in person (or no)
Yes, there is an online component or the meeting is held online (or not)

Thanks for your consideration,
Nicky

**Comment by tim-rohrer** [comment]
Hey Nicky, Thank you for the feedback. There is room for improvement in the plugin's feedback mechanisms, and so we'll discuss this internally. Toggles work for some people, but not all, and they can add complexity because validation of data isn't inherent in the processing (probably preferred). 

We've discussed having a checkbox to temporarily close meetings, but for the COVID situation, we determined the best way was to use the dialogue currently present on the edit meeting page rather than having the user try to figure out the different combinations of multiple switches. The user's answers here generate the necessary database values to track if a meeting is TC. I admit, I'm not clear how TSML would have interpreted your imported data as TC *unless* you had the type listed. Can you send me the csv (devs@code4recovery.org) you used for the import?

Online meetings are a good example of where complexity would be introduced if we had a checkbox on the edit meeting page. A meeting could be checked as online, but then we'd still need to validate if a URL was provided. So why bother with the checkbox? If no URL, the meeting *can't* be online; if the URL is present, the meeting can be held online. In this sense, the online meeting situation is different than the TC situation. 

Regrettably, the approach for TC is a little more complicated, and so the data is needed explicitly in the csv for import. There has to be some way of the user telling the code. As I mentioned above, if that indication wasn't in your CSV, then we need to look closer and *why* those six meetings were indicated as closed.

**Comment by OSAwebsite** [comment]
Hi Tim,

In this particular case the TC WAS in my checkboxes for types, so the error
was mine entirely.
But it is such an easy error to make, just because we don't have a separate
toggle in the import sheet as we have in edit mode.
I know for certain I am not the only one using the import sheet and my idea
is: when you offer an import sheet, make it as explicit and easy as
possible.

I am a database programmer in FileMaker myself, have been for over 30
years. It is not the same as the programming for the website, but I do
understand the issue of complexity.
More fields / entry boxes does not always equal complexity though. I have
created something in FileMaker just for the purpose of showing you what I
mean rather than trying to capture it in words.
If you are interested, do you have a personal email (use my email address
to send it to me)?
I can then send you a short screenvideo that might be helpful. Seems to
detailed to  include in this post.

Let me know....




-
Yours in service,
Nicky D
Acting Ontario South Website Coordinator
al-anon.alateen.on.ca

*This too will pass...*

**Reply by tim-rohrer** [reply]
If you can link the screenshot/video here, that would be great. Or send it to devs@code4recovery.org.

I would personally rather see a separate field for tracking TC, but that would mean discussions about modifying the spec, and I don't think we want to go there. 

As I said, we'll discuss internally and your mockup could help with that. Thank you.

**Comment by OSAwebsite** [comment]
OK, I have just sent it by email. Due to the size it will come as a
download link. Let me know if you have trouble getting it.

Changing specs in an existing tool is always a pain. I understand where you
are coming from. Maybe there are not that many people having trouble with
the tags either, I don't know.
But I do think that often it is worth the effort to at least consider
future developments through different glasses.
Thinking of the program principle of 'don't do for another what they can do
for themselves': if you give people the option to enter toggles, then not
only does it become more clear to them how TSML uses those settings, it
also makes coding easier to read. At least, that is how it is for me.

My first rendering of a complex procedure is never the best. Usually I just
want to get the code to do what I want. After that, most of the time I
return to it to simplify it and make it more modular, if that makes sense.

Good luck with this. If I can help, let me know. Even when I am no longer
the webcoordinator, I could install the tool on a private page on my own
personal website. I am always willing to be a tester, if you need one.

Cheers,
Nicky


-
Yours in service,
Nicky D
Acting Ontario South Website Coordinator
al-anon.alateen.on.ca

*This too will pass...*
